### PR TITLE
[Parameter] Prevent names from starting or ending with a colon

### PR DESCRIPTION
**Pull request by fancyweb**
As requested by @Majkl578 in #5996

### CODE REVIEW

**Comment by Ocramius** [comment]
@fancyweb can you check out `master`, cherry-pick https://github.com/doctrine/doctrine2/pull/6881/commits/d069ee3f85c54851dd272bd4e26b50c752671de3 into a new branch and force-push it here please?

**Comment by fancyweb** [comment]
@Ocramius Sorry for the delay, it's done.

File: tests/Doctrine/Tests/ORM/Query/ParameterTest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+namespace Doctrine\Tests\ORM\Query;
+
+use Doctrine\ORM\Query\Parameter;
+use PHPUnit\Framework\TestCase;
+
+class ParameterTest extends TestCase
+{
+    /**
+     * @dataProvider getParameterNamesStartingOrEndingWithAColon
+     * @expectedException \InvalidArgumentException
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Query/ParameterTest.php]: Please move these to explicit calls to `expect*()` in the test body

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by derrabus** [comment]
I'm removing this PR from the 3.0.0 milestone for now because it's blocked by #6880.

**Pull request closed without merging**